fluent/ExtrinsicRune.ts

```diff
@@ -46,20 +49,35 @@ export class ExtrinsicRune<out C extends Chain, out U> extends PatternRune<Chain
       .call(this.chain.into(ValueRune).access("metadata"))
       .into(CodecRune)
       .encoded(Rune.rec({
-        protocolVersion: 4,
+        protocolVersion: ExtrinsicRune.PROTOCOL_VERSION,
         call: this,
       }))
   }
 
   feeEstimate() {
-    const extrinsicHex = this.encoded().map(hex.encodePrefixed)
-    return this.chain.connection.call("payment_queryInfo", extrinsicHex)
-      .map(({ weight, ...rest }) => ({
-        ...rest,
-        weight: {
-          proofSize: BigInt(typeof weight === "number" ? 0 : weight.proof_size),
-          refTime: BigInt(typeof weight === "number" ? weight : weight.ref_time),
-        },
-      }))
+    const $queryInfoCallArgs = Rune.fn($extrinsic)
+      .call(this.chain.into(ValueRune).access("metadata"))
+      .map((ext) => $.tuple(ext, $.u32))
+
+    const args = $queryInfoCallArgs
+      .into(CodecRune)
+      .encoded(Rune.tuple([
+        Rune.rec({
+          protocolVersion: ExtrinsicRune.PROTOCOL_VERSION,
+          call: this,
+        }),
+        this.encoded().access("length"),
+      ]))
+
+    const $transactionPaymentApiQueryInfoResult = $.object(
```

Comment:
Is there a consistent path at which this type exists? If so, you could access the codec from the metadata's types instead of manually hard-coding its shape. `metadata.access("types", path)`.